core/primitives/src/views.rs

```diff
@@ -115,6 +125,30 @@ impl From<AccountView> for Account {
     }
 }
 
+impl From<&ContractCode> for ContractCodeView {
+    fn from(contract_code: &ContractCode) -> Self {
+        ContractCodeView { code: contract_code.code.clone(), hash: contract_code.hash.clone() }
+    }
+}
+
+impl From<ContractCode> for ContractCodeView {
+    fn from(contract_code: ContractCode) -> Self {
+        (&contract_code).into()
+    }
+}
```

Comment:
This will cause an unnecessary clone when you consume `contract_code`. I suggest we don't implement the `From<&T>` variant:

```suggestion
impl From<ContractCode> for ContractCodeView {
    fn from(contract_code: ContractCode) -> Self {
        ContractCodeView { code: contract_code.code, hash: contract_code.hash }
    }
}
```

When the `into` is needed for `&T`, you can do `x.clone().into()` or, if there is an iterator `xxx.into_iter().clonned().map(Into::into)`